### ISSUE TITLE
screenshots incorrectly attached to Allure report

### ISSUE THREAD

**Issue by drsens** [mmisty/cypress-allure-adapter#82]
Hi,

I'm using the latest version of the plugin i.e. 0.15.2 and I've encountered a few issues when working with screenshots (taken using native cy.screenshot() )

ISSUE#1  Screenshots get readded to consecutive tests

As an example test structure, I have:

* Test Suite 1
** Test 1
** Test 2
* Test Suite 2
** Test 1

I also have a common method which is used across all tests and takes a screenshot of an element I captured with cy.get
The problem is that when this is called in Test Suite 1 / Test 1 and then in Test 2 and further in Test Suite 2 / Test 1, it's all pilled together and by the time it reaches the test in Test Suite 2, there are 3 identical screenshots attached to that test.

So in the end I have:
* Test Suite 1
** Test 1 - screenshot.png
** Test 2 -  screenshot.png,  screenshot.png
* Test Suite 2
** Test 1 -  screenshot.png,  screenshot.png,  screenshot.png

ISSUE#2 Screenshots are added to the bottom of steps list instead of under specific step

This would be great if there was an option to have screenshots under specific steps, i.e. if I have method() that has a step inside its body as well as screenshot(), that screenshot would appear under the step in Allure report.

Thanks in advance for any feedback!

**Comment by mmisty** [comment]
@drsens hello! 
It would be really helpful if you could add test/spec examples so I could understand the issues better.

**Comment by drsens** [comment]
@mmisty surething, I'll try to put some code snippets, though not able to share the full repo example at the minute. Hopefully the below explains it better though!

Regarding **ISSUE#1**:
I'm using Page Object Model so I have a few different files representing objects in my website. One of them is the login page and it has a method like:
```typescript
submitLoginForm() {
        cy.allure().startStep('Submitting login form');
        this.SUBMIT_FORM().click();
        this.navigation.closeNavigation();
        cy.get('[aria-label="User profile"]').screenshot().should('be.visible');
        cy.allure().endStep();
}
```

Now in my test/spec files I have one that contains 3 tests, all of them calling the same method somewhere throughout the test:
```typescript
it('test 1', () => {
        ...
        loginPage.submitLoginForm();
});

it('test 2', () => {
        ...
        loginPage.submitLoginForm();
});

it('test 3', () => {
        ...
        loginPage.submitLoginForm();
});
```

If I run the full suite with 3 tests, the outcome I see is:

- test 1 has a single screenshot attached in allure report
- test 2 has two screenshots attached in allure report
- test 3 has three screenshots attached in allure report

The method with screenshot is only called ONCE per  each test so each test should only have ONE screenshot. What actually happens is that screenshots from previous tests also get added to the next tests, so each consecutive test has more and more screenshots if the method is called in next tests.

Regarding **ISSUE#2**
I'm using startStep/endStep like in `submitLoginForm()` method above, with screenshot taken inside the body. My expectation would be that this screenshot is attached under this specific step in Allure report (in this case 'Submitting login form'), but what happens is that all screenshots are at the very end of the report, under all steps.

I did some digging and testing and found out that it's most likely because in `cypress-allure-adapter/tree/main/src/setup/screenshots.ts` you are using:
`Cypress.Allure.testFileAttachment(basename(path), path, getContentType(path));`

and that is attaching all screenshots to the end of the steps list. Based on my tests, using:
`Cypress.Allure.fileAttachment` instead is correctly attaching screenshots under steps where these screenshots were taken. 

I also looked at the "screenshot fix" Pull Request that you have and you've also switched to using `Cypress.Allure.fileAttachment` there :) seems like that's the culprit of the issue.

Let me know if something is still unclear, thanks!

**Comment by mmisty** [comment]
@drsens
issue2: accepted. Yes, currently all screenshot attached to a test, not to a step. Will add setting to change that.
issue1: cannot reproduce: 
test
```typescript
describe('screenshot', () => {
  describe('screenshot @screen', () => {
    beforeEach(()=> {
      visitHtml();
    });
    
    it('test 1', () => {
      cy.allure().startStep('step 1');
      cy.screenshot();3
      cy.allure().endStep();
    });
    
    it('test 2', () => {
      cy.allure().startStep('step 1');
      cy.screenshot();
      cy.allure().endStep();
    });
    
    it('test 3', () => {
      cy.allure().startStep('step 1');
      cy.screenshot();
      cy.allure().endStep();
    });
  });
});
```
result: 
<img width="972" alt="image" src="https://github.com/mmisty/cypress-allure-adapter/assets/16957275/15cea120-bb32-458d-a002-3da485a0de83">

<img width="807" alt="image" src="https://github.com/mmisty/cypress-allure-adapter/assets/16957275/491a50a3-1143-46d4-b239-ce779b6fe860">

<img width="775" alt="image" src="https://github.com/mmisty/cypress-allure-adapter/assets/16957275/48b3c832-7cc7-4ebf-8275-53d7f817e6cf">

**Comment by drsens** [comment]
@mmisty good to know regarding issue#2, it will definitely make the reports look nicer once it's available in the upcoming version of the plugin.

As for issue#1, I think I managed to get to the bottom of the problem. It appears that the duplication occurs when you assign a filename to the screenshots taken, for example in my case it would be:
```typescript
cy.get('[aria-label="User profile"]').screenshot('nice screenshot').should('be.visible');
```
It's fine when the filenames are different, but since I'm reusing my single method in different tests, I get names like:
```
screenshot.png
screenshot (1).png
screenshot (2).png
```

In your example, try changing all your `cy.screenshot()` lines to `cy.screenshot('customName')` and rerun the tests. You should get 3 screenshots in your cypress/screenshots folder:
```
customName.png
customName (1).png
customName (2).png
```

but in allure-reports, you should get 6 screenshots instead of 3 and you should see the issue I was facing.

Please let me know when you'll have a moment to try and how that went :) many thanks!

**Comment by mmisty** [comment]
@drsens Ok thank you for letting me know. 
Now we know that there is workaround for the issue 1- you can add timestamp to screenshot names and it should be good.
Anyway I'll try to fix it.

**Comment by drsens** [comment]
@mmisty yeah that's exactly what I'm gonna do, I think I wanted to add timestamps to names anyways so it's okay 👍 

in which case the "attach screenshots to steps" fix is the one I would be waiting for :D

**Comment by mmisty** [comment]
@drsens both issues should be resolved in 0.16.0 version
